File: files/2025-06-22/CSEC 22062025.R
library(tidyverse)
library(readxl)

path = "files/2025-06-22/Challenge 36.xlsx"
input = read_excel(path, range = "B3:C7")
test  = read_excel(path, range = "E3:F11")

result = input %>%
  separate_longer_delim(Students, delim = ", ") 

all.equal(test, result, check.attributes = FALSE)
#> [1] TRUE

File: files/2025-06-22/CSEC 22062025.py
import pandas as pd

path = "files/2025-06-22/Challenge 36.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=2, nrows=4)
test = (
    pd.read_excel(path, usecols="E:F", skiprows=2, nrows=8)
      .rename(columns=lambda col: col.replace('.1', ''))
)

result = (
    input
    .assign(Student=input['Students'].str.split(', '))
    .explode('Student')
    .reset_index(drop=True)
    [['Student', 'Subject']]
)

print(result.equals(test))
